google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/SchemaCompact.java

```diff
@@ -76,24 +102,473 @@ private TableId getTableId(String tableName) {
+        || field == Descriptors.FieldDescriptor.Type.INT64
+        || field == Descriptors.FieldDescriptor.Type.SFIXED32
+        || field == Descriptors.FieldDescriptor.Type.SFIXED64) {
+
+      return true;
+    }
+    return false;
+  }
+
+  private static boolean isCompatibleWithBQDatetime(Descriptors.FieldDescriptor.Type field) {
+    if (field == Descriptors.FieldDescriptor.Type.INT64
+        || field == Descriptors.FieldDescriptor.Type.SFIXED64) {
+      return true;
+    }
+    return false;
+  }
+
+  private static boolean isCompatibleWithBQFloat(Descriptors.FieldDescriptor.Type field) {
+    if (field == Descriptors.FieldDescriptor.Type.FLOAT) {
+      return true;
+    }
+    if (field == Descriptors.FieldDescriptor.Type.DOUBLE) {
+      return true;
+    }
+    return false;
+  }
+
+  private static boolean isCompatibleWithBQGeography(Descriptors.FieldDescriptor.Type field) {
+    if (field == Descriptors.FieldDescriptor.Type.BYTES) {
+      return true;
+    }
+    return false;
+  }
+
+  private static boolean isCompatibleWithBQInteger(Descriptors.FieldDescriptor.Type field) {
+    if (field == Descriptors.FieldDescriptor.Type.INT64
+        || field == Descriptors.FieldDescriptor.Type.SFIXED64
+        || field == Descriptors.FieldDescriptor.Type.INT32
+        || field == Descriptors.FieldDescriptor.Type.UINT32
+        || field == Descriptors.FieldDescriptor.Type.FIXED32
+        || field == Descriptors.FieldDescriptor.Type.SFIXED32
+        || field == Descriptors.FieldDescriptor.Type.ENUM) {
+      return true;
+    }
+    return false;
+  }
+
+  private static boolean isCompatibleWithBQNumeric(Descriptors.FieldDescriptor.Type field) {
+    if (field == Descriptors.FieldDescriptor.Type.INT32
+        || field == Descriptors.FieldDescriptor.Type.INT64
+        || field == Descriptors.FieldDescriptor.Type.UINT32
+        || field == Descriptors.FieldDescriptor.Type.UINT64
+        || field == Descriptors.FieldDescriptor.Type.FIXED32
+        || field == Descriptors.FieldDescriptor.Type.FIXED64
+        || field == Descriptors.FieldDescriptor.Type.SFIXED32
+        || field == Descriptors.FieldDescriptor.Type.SFIXED64) {
+      return true;
+    }
+
+    if (field == Descriptors.FieldDescriptor.Type.BYTES) {
+      return true;
+    }
+
+    return false;
+  }
+
+  private static boolean isCompatibleWithBQRecord(Descriptors.FieldDescriptor.Type field) {
+    if (field == Descriptors.FieldDescriptor.Type.MESSAGE
+        || field == Descriptors.FieldDescriptor.Type.GROUP) {
+      return true;
+    }
+    return false;
+  }
+
+  private static boolean isCompatibleWithBQString(Descriptors.FieldDescriptor.Type field) {
+    if (field == Descriptors.FieldDescriptor.Type.STRING
+        || field == Descriptors.FieldDescriptor.Type.ENUM) {
+      return true;
+    }
+    return false;
+  }
+
+  private static boolean isCompatibleWithBQTime(Descriptors.FieldDescriptor.Type field) {
```

Comment:
also check format:

if (cpp_type == proto2::FieldDescriptor::CPPTYPE_INT64 &&
          (field_format == googlesql::FieldFormat::TIME_MICROS ||
           field_format == googlesql::FieldFormat::DEFAULT_FORMAT))

Reply:
Discussed with Yiru and we decided we are going to skip format checking for now!